pkg/api/query/v1.go

```diff
@@ -1185,18 +1204,31 @@ func (qapi *QueryAPI) series(r *http.Request) (interface{}, []error, *api.ApiErr
 		metrics = []labels.Labels{}
 		sets    []storage.SeriesSet
 	)
+
+	hints := &storage.SelectHints{
+		Limit: toHintLimit(limit),
+		Start: start.UnixMilli(),
+		End:   end.UnixMilli(),
+	}
+
 	for _, mset := range matcherSets {
-		sets = append(sets, q.Select(ctx, false, nil, mset...))
+		sets = append(sets, q.Select(ctx, false, hints, mset...))
 	}
 
 	set := storage.NewMergeSeriesSet(sets, storage.ChainedSeriesMerge)
+	warnings := set.Warnings()
 	for set.Next() {
 		metrics = append(metrics, set.At().Labels())
+		if limit > 0 && len(metrics) > limit {
+			metrics = metrics[:limit]
```

Comment:
If we pass the limit as a hint, why do we need to slice it here again? Or am I misunderstanding what `limit` represents here?

Reply:
It's just a hint. Backend could not implement it yet

Reply:
We should add it to store implementations as followup

Reply:
I see! Makes sense

Reply:
I will work on the store implementations next.